### PR TITLE
sv-4-users-save-to-database

### PR DESCRIPTION
**Pull request by svoggu**

### CODE REVIEW

File: src/client/src/app/components/register/register.component.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import { User } from '../../../../../shared/models/user.model';
+
+@Component({
+  selector: 'app-register',
+  templateUrl: './register.component.html',
+  styleUrls: ['./register.component.scss'],
+})
+export class RegisterComponent implements OnInit {
+  addUser: FormGroup;
+  selectedUser: User | null = null;
```
Comment (brycejjacks) [inline src/client/src/app/components/register/register.component.ts:19]: I don't understand why we have a selectedUser here?

File: src/client/src/app/components/register/register.component.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      password: [
+        '',
+        Validators.compose([Validators.required, Validators.minLength(5)]),
+      ],
+    });
+  }
+
+  ngOnInit(): void {}
+
+  postUser(selectedUser: User | null) {
```
Comment (brycejjacks) [inline src/client/src/app/components/register/register.component.ts:45]: personally not a fan of the name of this function or its parameter.... `registerUser(user: User | null)` is far more clear to me... BUTT i would not have a parameter on the function either way ... you can access the form inside this function with `this.adduser.value`

File: src/client/src/app/components/register/register.component.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    !selectedUser
```
Comment (brycejjacks) [inline src/client/src/app/components/register/register.component.ts:46]: i dont think selectedUser should have the option to be null.

**Review by brycejjacks — CHANGES_REQUESTED** [review]

**Review by brycejjacks — DISMISSED** [review]

File: src/client/src/app/components/register/register.component.html [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        </div>
+  
+        <div class="email_password">
+          <input type="text" formControlName="email" placeholder="email" />
+          
+          <input type="text" formControlName="password" placeholder="password" />
+        </div>
+  
+       
+       <button type="submit" class="submit-button" (click)="postUser(selectedUser)" [disabled]="addUser.invalid">
```
Comment (brycejjacks) [inline src/client/src/app/components/register/register.component.html:21]: still an at large issue

**Review by brycejjacks — DISMISSED** [review]

**Review by sabiyoung — APPROVED** [review]

File: src/client/src/app/components/register/register.component.html [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  
+    <div class="body">
+      <form [formGroup]="addUser">
+        <div class="user_name">
+          <input type="text" placeholder="Name" formControlName="name" />
+          <input type="text" placeholder="UserName" formControlName="username" />
+        </div>
+  
+        <div class="email_password">
+          <input type="text" formControlName="email" placeholder="email" />
```
Comment (svoggu) [inline src/client/src/app/components/register/register.component.html:15]: sure

**Pull request merged**